lib/resources/service.rb

```diff
@@ -134,9 +134,14 @@ def select_service_mgmt # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticCom
         else
           SysV.new(inspec, service_ctl || '/usr/sbin/service')
         end
-      elsif %w{redhat fedora centos}.include?(platform)
-        version = os[:release].to_i
-        if (%w{ redhat centos }.include?(platform) && version >= 7) || (platform == 'fedora' && version >= 15)
+      elsif os.redhat?
```

Comment:
Thanks @chris-rock, I've reverted to the previous commit - messy but it works :)